File: day6/day6.py
input = open('day6/input.txt','r')
lines = input.readlines()
input.close()
lines.append("")

qcount={}
rowcount=0
answer1=0
answer2=0
for line in lines:
  line=line.strip()
  if (line==""):
    answer1+=len(qcount)
    for (k,v) in qcount.items():
        if v==rowcount: answer2+=1 
    qcount={}
    rowcount=0
  else:
    rowcount+=1
    for c in line: 
        if (c in qcount.keys()): qcount[c]+=1
        else: qcount[c]=1
    
print(answer1,answer2)

File: day6/day6.hs
import System.IO
import Data.List (intersect,sort)
import Data.List.Split (splitOn)
import Data.String.Utils (strip)
import Data.List.Utils (uniq)

main = do
    input <- readFile "day6/input.txt"
    let records = splitOn "\n\n" $ input
    let answer1 =  map (length . (filter (/= '\n')). uniq) records
    let answer2 =  map (length . (foldr1 intersect) . lines) records
    print (sum answer1, sum answer2)

    